### PR TITLE
RichCommands: EnterCompactOverlay, ExitCompactOverlay, ToggleCompactOverlay

### LINKED ISSUE #11481: Generalizing Rich Command Usage
### Description

Thanks to @cinqmilleans' job, Files has now a cleaner way to handle commands across the solution. Rich commands now need to be implemented throughtout the whole solution.

This issue is used to keep track of which commands are made into Rich Commands.

### Concerned code

The following commands must be transformed into Rich Commands (See `BaseLayoutCommandsViewModel`, `ContextFlyoutItemHelper` and `ToolbarViewModel`):

- [x] Clear Selection (#11502)
- [x] Close Selected Tab (#11916)
- [x] Close Pane (#12002)
- [x] Compress Into Archive (#11572)
- [x] Compress Into Seven Zip (#11572)
- [x] Compress Into Zip (#11572)
- [x] Copy Item (#11504)
- [x] Copy Path Of Selected Items (#11774)
- [ ] Create Folder With Selection
- [x] Create New File
- [x] Create New Folder (#11493)
- [x] Create Shortcut (#11493)
- [x] Create Shortcut From Dialog (#11493)
- [x] Cut Item (#11504)
- [x] Decompress Archive (#11723)
- [x] Decompress Archive Here (#11723)
- [x] Decompress Archive To Child Folder (#11723)
- [x] Delete Item (#11504)
- [x] Duplicate Tab (#11596)
- [x] Empty Recycle Bin (#11415 )
- [x] Enter compact overlay (#11821)
- [x] Exit compact overlay (#11821)
- [x] Format Drive #11831
- [x] Gridview Size Decrease (#11678)
- [x] Gridview Size Increase (#11678)
- [x] Group Layer Sort (#11678)
- [x] Install Font (#11780)
- [x] Install Inf Driver (#11812)
- [x] Invert Selection (#11502)
- [x] Multi-Selection (#11502)
- [x] Navigate back (#11829)
- [x] Navigate forward (#11829)
- [x] Navigate up (#11829)
- [x] Navigate To Next Tab (#11916)
- [x] - [x] Navigate To Previous Tab (#11916)
- [ ] Open Directory In New Pane
- [ ] Open Directory In New Tab
- [x] Open File Location
- [ ] Open In New Window Item
- [x] Open In Terminal (#11445)
- [x] Open Item (#11667)
- [x] Open Item With Application Picker (#11667)
- [x] Open New Tab (#11596)
- [x] Open Parent Folder (#11667)
- [x] Open new pane (#12002)
- [x] Open settings #11846
- [x] Paste Items From Clipboard (#11670)
- [x] Pin Directory To Favorite
- [x] Play All (#12017)
- [x] Quick Look (#11771)
- [x] Redo (#12017)
- [x] Refresh #11807
- [x] Rename Item #11861
- [x] Reopen Tab (#11833)
- [x] Restore Item (#11472)
- [x] Restore Selection Recycle Bin (#11472)
- [x] Rotate Image Left (#11633)
- [x] Rotate Image Right (#11633)
- [x] Run As Admin (#11581)
- [x] Run As Another User (#11581)
- [x] Run With Powershell (#11914)
- [x] Search Unindexed Items (#12074)
- [x] Select All (#11502)
- [ ] Send to
- [x] Set As Lockscreen Background (#11563)
- [x] Set As Slideshow Item(#11563)
- [x] Set as Desktop Background(#11563)
- [x] Show Folder Properties (#11915)
- [x] Show Properties (#11915)
- [ ] Show/hide dot files
- [x] Share file (#11827)
- [x] Sidebar Pin Item (#11492)
- [x] Sidebar Unpin Item (#11492)
- [x] Toggle Preview Pane (#11578)
- [x] Toggle Sidebar #11847
- [x] Undo (#12017)
- [x] Unpin Directory From Favorite
- [x] Unpin Item From Start

### Gains

- A more cohesive code.
- The ability to call commands easily from anywhere in the code.

### Requirements

- Each of the commands previously cited have been transformed into Rich Commands.
- The previous implementations of the commands are cleaned up.

### Comments

_No response_

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #11481 

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [x] Did you remove any strings from the en-us resource file?
   - [x] Did you search the solution to see if the string is still being used? 
- [x] Did you implement any design changes to an existing feature?
   - [x] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Open app
   2. Hit `F12` or `Ctrl+Alt+Up` or `Ctrl+Alt+Down` or use toolbar action

### CODE REVIEW

File: src/Files.App/Actions/Global/EnterCompactOverlayAction.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using Files.App.Extensions;
+using Microsoft.UI.Windowing;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Windows.Graphics;
+using Windows.System;
+
+namespace Files.App.Actions
+{
+	internal class EnterCompactOverlayAction : ObservableObject, IAction
```
Comment (yaira2) [inline src/Files.App/Actions/Global/EnterCompactOverlayAction.cs:14]: @QuaintMako should we be grouping all the compact overlay actions together or should they be separate?

File: src/Files.App/Strings/en-US/Resources.resw [hunk 2]
```diff
@@ -1782,10 +1782,10 @@
   <data name="NavigationToolbarClosePane.Label" xml:space="preserve">
     <value>Close pane</value>
   </data>
-  <data name="NavigationToolbarEnterCompactOverlay.Label" xml:space="preserve">
```
Comment (yaira2) [inline src/Files.App/Strings/en-US/Resources.resw:1785]: Can you rename this in the other files so that we don't lose the translations?

Comment (ferrariofilippo) [reply src/Files.App/Strings/en-US/Resources.resw:1785]: We use that only once. I'll check that again

Comment (yaira2) [reply src/Files.App/Strings/en-US/Resources.resw:1785]: I meant that other translation files, otherwise it needs to be retranslated.

Comment (ferrariofilippo) [reply src/Files.App/Strings/en-US/Resources.resw:1785]: Do I need to remove strings from all translations too?

Comment (yaira2) [reply src/Files.App/Strings/en-US/Resources.resw:1785]: No

File: src/Files.App/Strings/en-US/Resources.resw [hunk 3]
```diff
@@ -2655,4 +2655,13 @@
   <data name="PasteItemToSelectionDescription" xml:space="preserve">
     <value>Paste item(s) from clipboard to the selected folder</value>
   </data>
+  <data name="ToggleCompactOverlayDescription" xml:space="preserve">
+    <value>Toggle compact overlay mode</value>
```
Comment (yaira2) [inline src/Files.App/Strings/en-US/Resources.resw:2659]: I think we can take out "mode"

**Comment by ferrariofilippo** [comment]
I think the command itself is self-explanatory

**Comment by hishitetsu** [comment]
> I think the command itself is self-explanatory

Yes, labels and descriptions can be the same. However, the description should be such that it is completely understandable what the action does, while labels are not always so.

File: src/Files.App/Actions/Global/EnterCompactOverlayAction.cs (hunk 1 above)
Comment (QuaintMako) [reply src/Files.App/Actions/Global/EnterCompactOverlayAction.cs:14]: I'd say separated, they are slightly different in their objectives.

**Pull request merged**